cbmc/proofs/prepare.py

```diff
@@ -24,16 +24,52 @@
 
 import logging
 import os
+import pathlib
+import subprocess
 import sys
 import textwrap
-from subprocess import CalledProcessError
 
 from make_cbmc_batch_files import create_cbmc_yaml_files
 
+
+def apply_patches():
+    patch_dir = pathlib.Path(__file__).resolve().parent.parent / "patches"
+    if not patch_dir.is_dir():
+        logging.error("Patches directory at '%s' is not a directory", patch_dir)
+        sys.exit(1)
+
+    proj_dir = pathlib.Path(__file__).resolve().parent.parent.parent
+    if not proj_dir.is_dir():
+        logging.error("Root of project at '%s' is not a directory", proj_dir)
+        sys.exit(1)
+    if not (proj_dir / "LICENSE").exists():
+        logging.error(
+            "Directory '%s' doesn't seem to be root of project", proj_dir)
+        sys.exit(1)
+
+    for fyle in sorted(patch_dir.glob("*.patch")):
+        logging.info("Checking patch '%s'", fyle.name)
+        cmd = ["git", "apply", "--check", str(fyle)]
+        proc = subprocess.run(
+            cmd, cwd=proj_dir, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+            text=True)
```

Comment:
This is not 3.5.  Consider newlines=True.

Reply:
done